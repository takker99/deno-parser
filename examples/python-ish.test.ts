import { assertSnapshot } from "@std/testing/snapshot";
import { Python } from "./python-ish.ts";

Deno.test("py complex", (t) =>
  assertSnapshot(
    t,
    Python.parse(
      `\
block:
  alpha
  bravo
  block:\r
          charlie
          delta\r
          echo
          block:
           foxtrot
  golf\
`,
    ),
  ));

Deno.test("py simple", (t) =>
  assertSnapshot(
    t,
    Python.parse(
      `\
block:
  alpha
  bravo
`,
    ),
  ));

Deno.test("py even simpler", (t) => assertSnapshot(t, Python.parse(`alpha`)));

Deno.test("py bad indent", (t) =>
  assertSnapshot(
    t,
    Python.parse(
      `\
block:
        alpha
        block:
    beta
`,
    ),
  ));
