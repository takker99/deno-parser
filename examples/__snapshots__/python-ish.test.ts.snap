export const snapshot = {};

snapshot[`py complex 1`] = `
{
  type: "ParseOK",
  value: {
    statements: [
      {
        type: "Ident",
        value: "alpha",
      },
      {
        type: "Ident",
        value: "bravo",
      },
      {
        statements: [
          {
            type: "Ident",
            value: "charlie",
          },
          {
            type: "Ident",
            value: "delta",
          },
          {
            type: "Ident",
            value: "echo",
          },
          {
            statements: [
              {
                type: "Ident",
                value: "foxtrot",
              },
            ],
            type: "Block",
          },
        ],
        type: "Block",
      },
      {
        type: "Ident",
        value: "golf",
      },
    ],
    type: "Block",
  },
}
`;

snapshot[`py simple 1`] = `
{
  type: "ParseOK",
  value: {
    statements: [
      {
        type: "Ident",
        value: "alpha",
      },
      {
        type: "Ident",
        value: "bravo",
      },
    ],
    type: "Block",
  },
}
`;

snapshot[`py even simpler 1`] = `
{
  type: "ParseOK",
  value: {
    type: "Ident",
    value: "alpha",
  },
}
`;

snapshot[`py bad indent 1`] = `
{
  expected: [
    "more than 4 spaces",
  ],
  location: {
    column: 5,
    index: 40,
    line: 4,
  },
  type: "ParseFail",
}
`;
